app/models/project.rb

```diff
@@ -38,6 +38,9 @@ class Project < ApplicationRecord
   has_one :readme, through: :repository
   has_many :repository_maintenance_stats, through: :repository
 
+  scope :least_recently_updated_stats, -> { joins(:repository_maintenance_stats).group('projects.id').where.not(repository: nil).order(Arel.sql('max(repository_maintenance_stats.updated_at) ASC')) }
+  scope :no_existing_stats, -> { includes(:repository_maintenance_stats).where(repository_maintenance_stats: {id: nil}).where.not(repository: nil) }
```

Comment:
This query is meant to find projects that have a repository with no stats at all.

Reply:
SQL

```
SELECT project.*, repository_maintenance_stats.* 
FROM projects 
LEFT OUTER JOIN repositories ON repositories.id = projects.repository_id 
LEFT OUTER JOIN repository_maintenance_stats ON repository_maintenance_stats.repository_id = repositories.id 
WHERE repository_maintenance_stats.id IS NULL AND projects.repository_id IS NOT NULL
```

It is not actually selecting project.* and repository_maintenance_stats.* but is selecting each individual column as an alias, but I think it is effectively every column.

Reply:
Explain plan:

```
QUERY PLAN
--------------------------------------------------------------------------------------------------------------------
 Hash Right Join  (cost=1101.48..1829.08 rows=1 width=465)
   Hash Cond: (repository_maintenance_stats.repository_id = repositories.id)
   Filter: (repository_maintenance_stats.id IS NULL)
   ->  Seq Scan on repository_maintenance_stats  (cost=0.00..610.24 rows=27024 width=56)
   ->  Hash  (cost=1082.73..1082.73 rows=1500 width=413)
         ->  Merge Right Join  (cost=207.23..1082.73 rows=1500 width=413)
               Merge Cond: (repositories.id = projects.repository_id)
               ->  Index Only Scan using repositories_pkey on repositories  (cost=0.29..1483.00 rows=25298 width=4)
               ->  Sort  (cost=206.94..210.69 rows=1500 width=409)
                     Sort Key: projects.repository_id
                     ->  Seq Scan on projects  (cost=0.00..127.81 rows=1500 width=409)
                           Filter: (repository_id IS NOT NULL)
```

Reply:
Similarly the repository_id is not null bit looks like it could get large